Add Index Alias for Cloud Security Latest Findings Index

CohenIdo opened the pull request:
solves:
- https://github.com/elastic/kibana/issues/187435
## Summary
Starting with the upcoming version of the Cloud Security Posture integration, we are introducing a new Latest Transform. This transform will be configured through the integration’s assets and will write findings to a new “latest” index.

## Solution
In this PR, I replaced all occurrences of our latest findings index with an index alias.
The alias used is the **same one** that will be introduced in the upcoming integration version: `security_solution-cloud_security_posture.misconfiguration_latest`.

For integration versions earlier than 2.00, we will manually create an index alias for the deprecated latest index `logs-cloud_security_posture.findings_latest-default`.
For integration versions 2.00 and above, the index alias will be automatically created or updated as part of the Transform setup.

This approach ensures we always reference the correct index using a consistent alias, depending on the Kibana and Integration state. See the table below for details:


| Scenario | Behavior |
|----------|----------|
| **Existing users upgrading their Kibana cluster** | A new index alias will be created, replacing references to the latest findings index. On subsequent upgrades, the alias will automatically point to the correct index behind the scenes. |
| **New users (no integration installed) upgrading Kibana** | The index alias will **not** be installed during the Kibana upgrade. It will be added later when the integration is installed as part of its assets. |
| **Serverless users installing the integration after the Kibana change but before the integration update** | The index alias will point to the old latest findings index. |

## Upgrade Test Step by Step
### **⚠️ Note: This PR will only work in conjunction with the following Elasticsearch change: [elastic/elasticsearch#128893](https://github.com/elastic/elasticsearch/pull/128893)**

1. Install the CSPM integration with a version earlier than 2.00.
2. Index some findings, then verify that the latest index is populated and that data appears on both the Findings and Dashboard pages.
3. Go to the Integrations page and enable the Display beta integrations toggle.
4. Navigate to the Cloud Security Posture integration, go to Settings, and upgrade to the latest version.
5. Confirm that the new index `security_solution-cloud_security_posture.misconfiguration_latest` is populated (wait for the transform’s first cycle to complete).
6. Delete the data from the `logs-cloud_security_posture.findings_latest-default` index.
7. Open the Findings page.
8. Check the score index and verify that new entries have been indexed.

### Out of Scope:
- [ ] Transform deprecation. 
- [ ] Data view's index pattern update.

CohenIdo committed:
first working version

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
fix type

CohenIdo committed:
fix typo

CohenIdo committed:
adjusrment

kfirpeled commented:
![Screenshot 2025-06-05 at 10 55 00](https://github.com/user-attachments/assets/48d4bd3c-b03c-42dc-a336-407141224302)

We can update the previous transform name/description to mention that it is `legacy`? I think it will help for anyone looking on this page to understand better which one is more relevant to look at

CohenIdo commented:
> We can update the previous transform name/description to mention that it is legacy? I think it will help for anyone looking on this page to understand better which one is more relevant to look at

@kfirpeled, I added it as action item to:
https://github.com/elastic/security-team/issues/12757

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
fix code review comments

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306023)
* Commit: 52941a48cc11df1f157c6bdbd00d0899166f787d

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/306023#01974330-5d3e-449d-a49f-9fbfba16e1bc)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306023","number":306023,"commit":"52941a48cc11df1f157c6bdbd00d0899166f787d"}],"number":306023}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/kibana-cloud-security-posture (Team:Cloud Security)

CohenIdo committed:
fix ci errors

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
ci fix

CohenIdo committed:
ci fix

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306510)
* Commit: 29ed728705241cc2e60c39275ad3cf0e992aae59

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/306510#019753ab-f564-4cd7-8f5c-990cbff3cd7a)

### History
* :broken_heart: [Build #306493](https://buildkite.com/elastic/kibana-pull-request/builds/306493) failed a052c3cb9d64c74f956eb0d9aaf83b073b261eff
* :broken_heart: [Build #306487](https://buildkite.com/elastic/kibana-pull-request/builds/306487) failed 285ff6217893c088d760b6eea2b402ca16b81118
* :broken_heart: [Build #306477](https://buildkite.com/elastic/kibana-pull-request/builds/306477) failed b8c991854d16b601fca421362babf49e1fc6ef74
* :broken_heart: [Build #306025](https://buildkite.com/elastic/kibana-pull-request/builds/306025) failed aba42acec5a10d9ee9d947e24c8cae563c2de6fa
* :broken_heart: [Build #306023](https://buildkite.com/elastic/kibana-pull-request/builds/306023) failed 52941a48cc11df1f157c6bdbd00d0899166f787d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306510","number":306510,"commit":"29ed728705241cc2e60c39275ad3cf0e992aae59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306493","number":306493,"commit":"a052c3cb9d64c74f956eb0d9aaf83b073b261eff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306487","number":306487,"commit":"285ff6217893c088d760b6eea2b402ca16b81118"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306477","number":306477,"commit":"b8c991854d16b601fca421362babf49e1fc6ef74"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306025","number":306025,"commit":"aba42acec5a10d9ee9d947e24c8cae563c2de6fa"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306023","number":306023,"commit":"52941a48cc11df1f157c6bdbd00d0899166f787d"}],"number":306510}
buildkite-pr-comment-->

CohenIdo committed:
ci fix

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
tests update

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307349)
* Commit: 605e4d2f9318cab6e150e394243c0b6869d3b1fc

### Failed CI Steps
* [Cloud Security Posture Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307349/jobs/01975e9c-2e0f-4818-b87d-d477527ef3f3/artifacts/01975ecb-c4bc-4bb5-b5cd-81766a78a948) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307349/jobs/01975e9c-2e0f-4818-b87d-d477527ef3f3/artifacts/01975ecb-c4bc-4846-8a70-1f06930e5c08) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings but host name is not the same as alert host name "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307349/jobs/01975e9c-2e0f-4818-b87d-d477527ef3f3/artifacts/01975ecb-c4bb-4b9a-895f-5385a54179e5) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307349/jobs/01975e9c-2e0f-4818-b87d-d477527ef3f3/artifacts/01975ecb-c4bc-4c4f-8786-2ea5d75988b1) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout User name - Has misconfiguration findings "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307349#01975e9c-2e0f-4818-b87d-d477527ef3f3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307349/jobs/01975e9c-2e0f-4818-b87d-d477527ef3f3/artifacts/01975ecb-c4bc-44a5-9e8f-5234b8bc2845) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout User name - Has misconfiguration findings but host name is not the same as alert host name "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"

### History
* :broken_heart: [Build #306516](https://buildkite.com/elastic/kibana-pull-request/builds/306516) failed ac6f09c4082f17ef840974adeb9352dfc157ea36
* :broken_heart: [Build #306510](https://buildkite.com/elastic/kibana-pull-request/builds/306510) failed 29ed728705241cc2e60c39275ad3cf0e992aae59
* :broken_heart: [Build #306493](https://buildkite.com/elastic/kibana-pull-request/builds/306493) failed a052c3cb9d64c74f956eb0d9aaf83b073b261eff
* :broken_heart: [Build #306487](https://buildkite.com/elastic/kibana-pull-request/builds/306487) failed 285ff6217893c088d760b6eea2b402ca16b81118
* :broken_heart: [Build #306477](https://buildkite.com/elastic/kibana-pull-request/builds/306477) failed b8c991854d16b601fca421362babf49e1fc6ef74

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307349","number":307349,"commit":"605e4d2f9318cab6e150e394243c0b6869d3b1fc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306516","number":306516,"commit":"ac6f09c4082f17ef840974adeb9352dfc157ea36"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306510","number":306510,"commit":"29ed728705241cc2e60c39275ad3cf0e992aae59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306493","number":306493,"commit":"a052c3cb9d64c74f956eb0d9aaf83b073b261eff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306487","number":306487,"commit":"285ff6217893c088d760b6eea2b402ca16b81118"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306477","number":306477,"commit":"b8c991854d16b601fca421362babf49e1fc6ef74"}],"number":307349}
buildkite-pr-comment-->

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
tests fix

CohenIdo committed:
tests fix

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308137)
* Commit: d7963c6ffad88b547b8930d0cb988eb034cc4a4f

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/308137#01976648-a25b-4e07-84f6-7e58c386019a)

### History
* :broken_heart: [Build #307988](https://buildkite.com/elastic/kibana-pull-request/builds/307988) failed 1fba97765b003473008fc5ecd202edbb64b4782d
* :broken_heart: [Build #307349](https://buildkite.com/elastic/kibana-pull-request/builds/307349) failed 605e4d2f9318cab6e150e394243c0b6869d3b1fc
* :broken_heart: [Build #306516](https://buildkite.com/elastic/kibana-pull-request/builds/306516) failed ac6f09c4082f17ef840974adeb9352dfc157ea36
* :broken_heart: [Build #306510](https://buildkite.com/elastic/kibana-pull-request/builds/306510) failed 29ed728705241cc2e60c39275ad3cf0e992aae59
* :broken_heart: [Build #306493](https://buildkite.com/elastic/kibana-pull-request/builds/306493) failed a052c3cb9d64c74f956eb0d9aaf83b073b261eff

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308137","number":308137,"commit":"d7963c6ffad88b547b8930d0cb988eb034cc4a4f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307988","number":307988,"commit":"1fba97765b003473008fc5ecd202edbb64b4782d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307349","number":307349,"commit":"605e4d2f9318cab6e150e394243c0b6869d3b1fc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306516","number":306516,"commit":"ac6f09c4082f17ef840974adeb9352dfc157ea36"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306510","number":306510,"commit":"29ed728705241cc2e60c39275ad3cf0e992aae59"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306493","number":306493,"commit":"a052c3cb9d64c74f956eb0d9aaf83b073b261eff"}],"number":308137}
buildkite-pr-comment-->

CohenIdo committed:
update test

maxcold reviewed (approved):
Code loos good, also tested different scenarious, didn't notice anything weird. 
One thing I realised that it's possible to have two integrations of different versions (1.13.1 and 2.0 for example) on the same stack, but I think it shouldn't be an issue. In the end cloudbeat ingests data in the same source index and then the transform should pick data up from both integrations

opauloh commented on x-pack/solutions/security/plugins/cloud_security_posture/server/plugin.ts:
```diff
@@ -209,12 +215,54 @@ export class CspPlugin
 
+  // For integration versions earlier than 2.00, we will manually create an index alias for the deprecated latest index 'logs-cloud_security_posture.findings_latest-default'.
+  // For integration versions 2.00 and above, the index alias will be automatically created or updated as part of the Transform setup.
+  intializeIndexAlias = async (esClient: ElasticsearchClient, logger: Logger): Promise<void> => {
```
small typo

```suggestion
  initializeIndexAlias = async (esClient: ElasticsearchClient, logger: Logger): Promise<void> => {
```

opauloh commented on x-pack/solutions/security/plugins/cloud_security_posture/server/plugin.ts:
```diff
@@ -209,12 +215,54 @@ export class CspPlugin
     await initializeCspIndices(esClient, this.config, this.logger);
     await initializeCspTransforms(esClient, this.logger);
     await scheduleFindingsStatsTask(taskManager, this.logger);
+    await this.intializeIndexAlias(esClient, this.logger);
```
small typo

```suggestion
    await this.initializeIndexAlias(esClient, this.logger);
```

opauloh commented on x-pack/solutions/security/plugins/cloud_security_posture/server/plugin.ts:
```diff
@@ -209,12 +215,54 @@ export class CspPlugin
+          ],
+        });
+        this.logger.info(
+          `Index alias ${CDR_LATEST_NATIVE_MISCONFIGURATIONS_INDEX_ALIAS} created successfully}`
```
nit:

```suggestion
          `Index alias ${CDR_LATEST_NATIVE_MISCONFIGURATIONS_INDEX_ALIAS} created successfully`
```

opauloh reviewed (approved):
Code LGTM! 💯 just small typos

CohenIdo committed:
update test

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

CohenIdo committed:
code review comments

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308550)
* Commit: fbb76805c40a7bc96226dd4fffb2594c6bb0a038

### Failed CI Steps
* [Cloud Security Posture Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308550/jobs/01977039-1d0f-45bd-a461-83a13bbeb3db/artifacts/01977067-956e-4039-a9a8-4926bcd5c5d2) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308550/jobs/01977039-1d0f-45bd-a461-83a13bbeb3db/artifacts/01977067-956d-40b4-944c-b6ceb7788e35) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings but host name is not the same as alert host name "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308550/jobs/01977039-1d0f-45bd-a461-83a13bbeb3db/artifacts/01977067-956e-44e2-be1a-08929d85bf0a) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout Host name - Has misconfiguration findings should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308550/jobs/01977039-1d0f-45bd-a461-83a13bbeb3db/artifacts/01977067-956f-49dd-80c3-263a1eca8f2b) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout User name - Has misconfiguration findings "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/308550#01977039-1d0f-45bd-a461-83a13bbeb3db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/308550/jobs/01977039-1d0f-45bd-a461-83a13bbeb3db/artifacts/01977067-956e-496b-8e65-c9bf9a31ecd4) Cloud Security Posture Cypress Tests / Alert Host details expandable flyout User name - Has misconfiguration findings but host name is not the same as alert host name "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings" "after each" hook for "should display Misconfiguration preview under Insights Entities when it has Misconfiguration Findings"

### History
* :broken_heart: [Build #308137](https://buildkite.com/elastic/kibana-pull-request/builds/308137) failed d7963c6ffad88b547b8930d0cb988eb034cc4a4f
* :broken_heart: [Build #307988](https://buildkite.com/elastic/kibana-pull-request/builds/307988) failed 1fba97765b003473008fc5ecd202edbb64b4782d
* :broken_heart: [Build #307349](https://buildkite.com/elastic/kibana-pull-request/builds/307349) failed 605e4d2f9318cab6e150e394243c0b6869d3b1fc
* :broken_heart: [Build #306516](https://buildkite.com/elastic/kibana-pull-request/builds/306516) failed ac6f09c4082f17ef840974adeb9352dfc157ea36
* :broken_heart: [Build #306510](https://buildkite.com/elastic/kibana-pull-request/builds/306510) failed 29ed728705241cc2e60c39275ad3cf0e992aae59
* :broken_heart: [Build #306493](https://buildkite.com/elastic/kibana-pull-request/builds/306493) failed a052c3cb9d64c74f956eb0d9aaf83b073b261eff

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308550","number":308550,"commit":"fbb76805c40a7bc96226dd4fffb2594c6bb0a038"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308137","number":308137,"commit":"d7963c6ffad88b547b8930d0cb988eb034cc4a4f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307988","number":307988,"commit":"1fba97765b003473008fc5ecd202edbb64b4782d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307349","number":307349,"commit":"605e4d2f9318cab6e150e394243c0b6869d3b1fc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306516","number":306516,"commit":"ac6f09c4082f17ef840974adeb9352dfc157ea36"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306510","number":306510,"commit":"29ed728705241cc2e60c39275ad3cf0e992aae59"}],"number":308550}
buildkite-pr-comment-->

CohenIdo committed:
update tests

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308563)
* Commit: 7b38d5b739cf1b49288d54a92977d99fb6143404
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222567-7b38d5b739cf`

### History
* :broken_heart: [Build #308550](https://buildkite.com/elastic/kibana-pull-request/builds/308550) failed fbb76805c40a7bc96226dd4fffb2594c6bb0a038
* :broken_heart: [Build #308137](https://buildkite.com/elastic/kibana-pull-request/builds/308137) failed d7963c6ffad88b547b8930d0cb988eb034cc4a4f
* :broken_heart: [Build #307988](https://buildkite.com/elastic/kibana-pull-request/builds/307988) failed 1fba97765b003473008fc5ecd202edbb64b4782d
* :broken_heart: [Build #307349](https://buildkite.com/elastic/kibana-pull-request/builds/307349) failed 605e4d2f9318cab6e150e394243c0b6869d3b1fc
* :broken_heart: [Build #306516](https://buildkite.com/elastic/kibana-pull-request/builds/306516) failed ac6f09c4082f17ef840974adeb9352dfc157ea36

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308563","number":308563,"commit":"7b38d5b739cf1b49288d54a92977d99fb6143404"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308550","number":308550,"commit":"fbb76805c40a7bc96226dd4fffb2594c6bb0a038"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308137","number":308137,"commit":"d7963c6ffad88b547b8930d0cb988eb034cc4a4f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307988","number":307988,"commit":"1fba97765b003473008fc5ecd202edbb64b4782d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307349","number":307349,"commit":"605e4d2f9318cab6e150e394243c0b6869d3b1fc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306516","number":306516,"commit":"ac6f09c4082f17ef840974adeb9352dfc157ea36"}],"number":308563}
buildkite-pr-comment-->

CohenIdo committed:
test update

CohenIdo committed:
fix test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308593)
* Commit: 5ccf119ab9d5985b5a080401098cfa9effec071a
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222567-5ccf119ab9d5`

### History
* :broken_heart: [Build #308569](https://buildkite.com/elastic/kibana-pull-request/builds/308569) failed 85e046ebeae2f74481feae6baff0026193ba6cf8
* :broken_heart: [Build #308563](https://buildkite.com/elastic/kibana-pull-request/builds/308563) failed 7b38d5b739cf1b49288d54a92977d99fb6143404
* :broken_heart: [Build #308550](https://buildkite.com/elastic/kibana-pull-request/builds/308550) failed fbb76805c40a7bc96226dd4fffb2594c6bb0a038
* :broken_heart: [Build #308137](https://buildkite.com/elastic/kibana-pull-request/builds/308137) failed d7963c6ffad88b547b8930d0cb988eb034cc4a4f
* :broken_heart: [Build #307988](https://buildkite.com/elastic/kibana-pull-request/builds/307988) failed 1fba97765b003473008fc5ecd202edbb64b4782d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308593","number":308593,"commit":"5ccf119ab9d5985b5a080401098cfa9effec071a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308569","number":308569,"commit":"85e046ebeae2f74481feae6baff0026193ba6cf8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308563","number":308563,"commit":"7b38d5b739cf1b49288d54a92977d99fb6143404"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308550","number":308550,"commit":"fbb76805c40a7bc96226dd4fffb2594c6bb0a038"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308137","number":308137,"commit":"d7963c6ffad88b547b8930d0cb988eb034cc4a4f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307988","number":307988,"commit":"1fba97765b003473008fc5ecd202edbb64b4782d"}],"number":308593}
buildkite-pr-comment-->

CohenIdo committed:
tests update

CohenIdo committed:
Merge remote-tracking branch 'upstream/main' into create-index-alias

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

opauloh committed:
pointing CDR misconfiguration tests to latest misconfiguration index

CohenIdo committed:
Merge branch 'main' into create-index-alias

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309152)
* Commit: d3832058372ae85a814847cbc49ef4bb7517bb5d

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/309152#01977c3d-9b8c-4b78-b33d-3ebbc827e98f)

### History
* :broken_heart: [Build #308990](https://buildkite.com/elastic/kibana-pull-request/builds/308990) failed 1f6a5aa65cc913f33aa1d81cd328b4c33854af8e
* :broken_heart: [Build #308915](https://buildkite.com/elastic/kibana-pull-request/builds/308915) failed daee1c9721dda5c1a7cca64adec29240f566f868
* :broken_heart: [Build #308593](https://buildkite.com/elastic/kibana-pull-request/builds/308593) failed 5ccf119ab9d5985b5a080401098cfa9effec071a
* :broken_heart: [Build #308569](https://buildkite.com/elastic/kibana-pull-request/builds/308569) failed 85e046ebeae2f74481feae6baff0026193ba6cf8
* :broken_heart: [Build #308563](https://buildkite.com/elastic/kibana-pull-request/builds/308563) failed 7b38d5b739cf1b49288d54a92977d99fb6143404

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309152","number":309152,"commit":"d3832058372ae85a814847cbc49ef4bb7517bb5d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308990","number":308990,"commit":"1f6a5aa65cc913f33aa1d81cd328b4c33854af8e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308915","number":308915,"commit":"daee1c9721dda5c1a7cca64adec29240f566f868"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308593","number":308593,"commit":"5ccf119ab9d5985b5a080401098cfa9effec071a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308569","number":308569,"commit":"85e046ebeae2f74481feae6baff0026193ba6cf8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308563","number":308563,"commit":"7b38d5b739cf1b49288d54a92977d99fb6143404"}],"number":309152}
buildkite-pr-comment-->

opauloh committed:
fix CI

opauloh committed:
fixing FTR Config #110

Fixing [job] [logs] FTR Configs #110 / Cloud Security Posture Old Data Findings page with old data returns no Findings KSPM

CohenIdo committed:
fix ci error

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309714)
* Commit: 4c7efbbd8c6c296d2af186633cbcb41cd738a65b
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222567-4c7efbbd8c6c`

### History
* :broken_heart: [Build #309686](https://buildkite.com/elastic/kibana-pull-request/builds/309686) failed 5ad526ab617a88619c5cd72e7bc08baecbad9841
* :broken_heart: [Build #309168](https://buildkite.com/elastic/kibana-pull-request/builds/309168) failed 80e4d8a49d55f084d1ca1610acbc2a9c33d112cb
* :broken_heart: [Build #309152](https://buildkite.com/elastic/kibana-pull-request/builds/309152) failed d3832058372ae85a814847cbc49ef4bb7517bb5d
* :broken_heart: [Build #308990](https://buildkite.com/elastic/kibana-pull-request/builds/308990) failed 1f6a5aa65cc913f33aa1d81cd328b4c33854af8e
* :broken_heart: [Build #308915](https://buildkite.com/elastic/kibana-pull-request/builds/308915) failed daee1c9721dda5c1a7cca64adec29240f566f868

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309714","number":309714,"commit":"4c7efbbd8c6c296d2af186633cbcb41cd738a65b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309686","number":309686,"commit":"5ad526ab617a88619c5cd72e7bc08baecbad9841"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309168","number":309168,"commit":"80e4d8a49d55f084d1ca1610acbc2a9c33d112cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309152","number":309152,"commit":"d3832058372ae85a814847cbc49ef4bb7517bb5d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308990","number":308990,"commit":"1f6a5aa65cc913f33aa1d81cd328b4c33854af8e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308915","number":308915,"commit":"daee1c9721dda5c1a7cca64adec29240f566f868"}],"number":309714}
buildkite-pr-comment-->

CohenIdo committed:
fix ci error

CohenIdo committed:
fix ci error

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309722)
* Commit: 7066439ebf0cade25e548e948bf5c467d18f654e

### Failed CI Steps
* [Mark CI Stats as ready](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781dc-7fce-4cac-aaa6-54a01d013b5e)
* [Mark CI Stats as ready](https://buildkite.com/elastic/kibana-pull-request/builds/309722#01978212-c374-462e-b3af-4a0885a5ec51)
* [FTR Configs #18](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a086-4401-a863-c2b74bd4b460)
* [FTR Configs #21](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a088-49e5-b3d5-2d85c38a2800)
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a08f-4dbc-94c2-7d2f86c089a4)
* [FTR Configs #35](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a093-4198-a576-42c21d417790)
* [FTR Configs #43](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a09a-460d-9018-17b4d146ce23)
* [FTR Configs #49](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a09e-4ee9-92aa-93baa7959f34)
* [FTR Configs #75](https://buildkite.com/elastic/kibana-pull-request/builds/309722#01978212-d5b8-471c-b1d2-ed95f5be98e9)
* [FTR Configs #83](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a0d3-4891-9948-440fe8cd75a9)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a0d4-430f-9dd4-26bf298209a5)
* [FTR Configs #85](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a0d5-4798-830b-e3a175107a30)
* [FTR Configs #115](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a0f9-43e2-9530-776833e9329f)
* [FTR Configs #118](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a0fc-4985-999c-0ae7c198b8b5)
* [FTR Configs #132](https://buildkite.com/elastic/kibana-pull-request/builds/309722#019781f5-a109-46c3-a486-ece4feb28b57)

### History
* :green_heart: [Build #309714](https://buildkite.com/elastic/kibana-pull-request/builds/309714) succeeded 4c7efbbd8c6c296d2af186633cbcb41cd738a65b
* :broken_heart: [Build #309686](https://buildkite.com/elastic/kibana-pull-request/builds/309686) failed 5ad526ab617a88619c5cd72e7bc08baecbad9841
* :broken_heart: [Build #309168](https://buildkite.com/elastic/kibana-pull-request/builds/309168) failed 80e4d8a49d55f084d1ca1610acbc2a9c33d112cb
* :broken_heart: [Build #309152](https://buildkite.com/elastic/kibana-pull-request/builds/309152) failed d3832058372ae85a814847cbc49ef4bb7517bb5d
* :broken_heart: [Build #308990](https://buildkite.com/elastic/kibana-pull-request/builds/308990) failed 1f6a5aa65cc913f33aa1d81cd328b4c33854af8e

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309722","number":309722,"commit":"7066439ebf0cade25e548e948bf5c467d18f654e"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309714","number":309714,"commit":"4c7efbbd8c6c296d2af186633cbcb41cd738a65b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309686","number":309686,"commit":"5ad526ab617a88619c5cd72e7bc08baecbad9841"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309168","number":309168,"commit":"80e4d8a49d55f084d1ca1610acbc2a9c33d112cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309152","number":309152,"commit":"d3832058372ae85a814847cbc49ef4bb7517bb5d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308990","number":308990,"commit":"1f6a5aa65cc913f33aa1d81cd328b4c33854af8e"}],"number":309722}
buildkite-pr-comment-->

CohenIdo committed:
Merge branch 'main' into create-index-alias

CohenIdo committed:
solve conflicts

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309862)
* Commit: 123806a593ae049fcecc7e8af7ce72ff3fc1ea53
* Kibana Serverless Image: `docker.elastic.co/kibana-ci/kibana-serverless:pr-222567-123806a593ae`

### History
* :broken_heart: [Build #309722](https://buildkite.com/elastic/kibana-pull-request/builds/309722) failed 7066439ebf0cade25e548e948bf5c467d18f654e
* :green_heart: [Build #309714](https://buildkite.com/elastic/kibana-pull-request/builds/309714) succeeded 4c7efbbd8c6c296d2af186633cbcb41cd738a65b
* :broken_heart: [Build #309686](https://buildkite.com/elastic/kibana-pull-request/builds/309686) failed 5ad526ab617a88619c5cd72e7bc08baecbad9841
* :broken_heart: [Build #309168](https://buildkite.com/elastic/kibana-pull-request/builds/309168) failed 80e4d8a49d55f084d1ca1610acbc2a9c33d112cb
* :broken_heart: [Build #309152](https://buildkite.com/elastic/kibana-pull-request/builds/309152) failed d3832058372ae85a814847cbc49ef4bb7517bb5d

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309862","number":309862,"commit":"123806a593ae049fcecc7e8af7ce72ff3fc1ea53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309722","number":309722,"commit":"7066439ebf0cade25e548e948bf5c467d18f654e"},{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309714","number":309714,"commit":"4c7efbbd8c6c296d2af186633cbcb41cd738a65b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309686","number":309686,"commit":"5ad526ab617a88619c5cd72e7bc08baecbad9841"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309168","number":309168,"commit":"80e4d8a49d55f084d1ca1610acbc2a9c33d112cb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309152","number":309152,"commit":"d3832058372ae85a814847cbc49ef4bb7517bb5d"}],"number":309862}
buildkite-pr-comment-->
